pkg/transport/generic_transport_controller.go

```diff
@@ -183,6 +183,15 @@ func (c *genericTransportController) enqueueBinding(obj interface{}) {
 // More info on this pattern here:
 // https://github.com/kubernetes/community/blob/8cafef897a22026d42f5e5bb3f104febe7e29830/contributors/devel/controllers.md
 func (c *genericTransportController) handleWrappedObject(obj interface{}) {
+	switch obj.(type) {
+	case cache.DeletedFinalStateUnknown:
+		// TODO: if all manifest objects have been deleted while ITS was done,
```

Comment:
typo, was down

Reply:
It's a todo and nothing to do about testing.  If you don't think this is worth pursuing in the future, I'll remove it.

Reply:
the wrapped objects live in ITS. if ITS is down there is no way to delete these objects.

I assume you meant for the situation where the transport controller is down, or lost the connection to ITS temporarily. I get your point in the comment. 
we need to pay attention that we put a finalizer on manifestwork objects that gets removed only after the Binding is cleaned (there is a cleanup order). 
so the situation you're describing in the comment could happen if the following happens (ALL of them):
1. transport controller is down or lost connectivity to ITS.
2. someone is deleting the manifestwork objects manually from ITS.
3. that someone is also deleting manually the finalizer from the manifestwork objects.

if the transport controller lost the events cause it was down, upon restart it gets events for all living resources and will get the system to the desired state (it will get all events for both Binding + manifestwork living objects).

if all the above happens when the controller was running but temporarily didn't have connectivity to ITS, it will bring the system back to the desired state in the next reconciliation (eventual consistency).

to summarize - **I don't think we should try to solve this kind of situation**. 
exactly for this we have a finalizer which is technically a concept that's used for cleanup purposes (cleanup is typically done by the controller).
if a user manually deletes the object + finalizer when the controller don't have connectivity to ITS, that user has to take into consideration the consequences. 
when I as a user delete a finalizer manually from an object, I'm aware of the fact that some of the cleanup that should be done, won't be done correctly.

Reply:
@nirrozenbaum I have no issue with eventual consistency, it makes the kubernetes world goes round.  What I have an issue with is the 10 hours that it will take.  We can argue whether it is valid for a user to delete the manifestworks, and I think I'll agree with you that it isn't when using ks, but there is no need to remove the finalizer, manifest-work-cleanup,  since it is owned by OCM and will not be impacted by transport-controller disappearance.  But I agree this is an edge case and we don't have to solve it.  I'll remove the TODO.